src/remote-config/condition-evaluator-internal.ts

```diff
@@ -293,6 +293,9 @@ function compareNumbers(
   return predicateFn(actual < target ? -1 : actual > target ? 1 : 0);
 }
 
+// Max number of segments a numeric version can have. This is enforced by the server as well.
+const MAX_LENGTH = 5;
```

Comment:
Thinking: three segments deviates from the semver spec, but consistency w the backend is more important, since RC is cross-platform.